src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -603,10 +604,12 @@ class CosmasGoogleCloudService(
             latestVersion.patchesList.last().userName
         }
 
+        val latestFileId = if (latestVersion.fileId.isBlank()) fileId else latestVersion.fileId
```

Comment:
```suggestion
        // In most cases latestVersion.fileId == fileId
        // However, when file changes its id (e.g. because of Dropbox import), we can
        // read its latest blob thanks to fileIdChangeMap, but we can't return its old id in 
        // FileVersionInfo object corresponding to the latest version.
        // In reality this makes the latest version unavailable, while previous versions
        // are still available.
        val latestFileId = if (latestVersion.fileId.isBlank()) fileId else latestVersion.fileId
```

Reply:
I changed your comment a bit. About testing, `hardRestoring` checks that fileVersionList is correct for this case